File: client/src/lib/types/bill.ts
import type { PartyVoteResponseDTO } from "./party-vote";

export type Bill = {
    id: number;
    billTag: string;
    title: string;
    question: string;
    description: string;
    forExplanation: string;
    againstExplanation: string;
    url: string;
    votedAt: string;
    isPassed: boolean;
    billType: string;
    partyVotes: PartyVoteResponseDTO[];
};

File: server/Hvadskaljegstemme/Dtos/Bill.cs
using Hvadskaljegstemme.Models.Enums;

namespace Hvadskaljegstemme.Dtos;

public record Bill
(
    int Id,
    string BillTag,
    string Title,
    string Description,
    string Question,
    string ForExplanation,
    string AgainstExplanation,
    string Url,
    bool IsPassed,
    BillType BillType,
    DateOnly VotedAt,
    int? Order,
    IEnumerable<PartyVote> PartyVotes
);

public record PartyVote
(
    int PartyId,
    Vote Vote
);
